### PR TITLE
Provide rough documentation on widgets, see issue #26

### PR DESCRIPTION
**Pull request by alxflam**
Provide a documentation regarding how to contribute views and dialogs in Theia.

Explained are: dialogs and views using plain DOM manipulation and React, inversify, css variables and themes.

The documentation regarding views is far from being complete but i think it's at least a start.

### CODE REVIEW

File: src/docs/Views.mdx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+title: Views
+slug: views
+---
+
+# Views and Dialogs
+
+Theia does not enforce you to use any specific UI library. For most of Theia's current UI elements either plain DOM modifications or React is being used. In this article we'll show you how to implement UI elements with both of these techniques.
+
+## Creating a plain HTML Dialog
+Especially non-complex dialogs can be easily implemented without an UI library. For example, Theia's "About" dialog is an example for this category. Let's start by defining the Menu Contribution and add a new Menu item in the "Help" category. To achieve this, we need to create a `dialog.ts` file where we'll specify our dialog. A common base clase for dialogs is `AbstractDialog`, we're going to inherit from it. The constructor of AbstractDialog expects an instance of `DialogProps` which is carrying the dialogs title.
```
Comment (vince-fugnitto) [inline src/docs/Views.mdx]: typo - `clase` should be `class`

File: src/docs/Views.mdx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+Running this will work but actually it won't look pleasing. Our text is black and therefore hard to read with the dark theme. Theia is making use of CSS variables to define common colors, fonts and styles. The dark and light theme respectively declare these variables. So by using these common variables, we ensure that our UI looks good in both the dark and light theme (and possible further themes which declare these variables). There are two ways of adding CSS to our DOM elements: either by directly manipulating the DOM or by using a custom stylesheet. We'll use the first approach in this example and the latter one is being used in the view example.
+
+Let's add the base font and color to our root div element:
+```typescript
+    messageNode.setAttribute('style', 'font-family: var(--theia-ui-font-family); color: var(--theia-ui-font-color1);');
+```
+Now our Dialog is readable and we could further enhance it's appearance by adding custom CSS.
+
+For reference, you can find all the variables in [variables-dark.useable.css](https://github.com/theia-ide/theia/blob/master/packages/core/src/browser/style/variables-dark.useable.css) (or the respective css of the light theme).
```
Comment (vince-fugnitto) [inline src/docs/Views.mdx]: The link to the main Theia repo is outdated.
https://github.com/eclipse-theia/theia/blob/master/packages/core/src/browser/style/variables-dark.useable.css

**Comment by jankeromnes** [comment]
Hi @alex0711, many thanks for contributing to Theia's documentation! 💯

Just to make sure, are you planning to continue working on this Pull Request, or would you like someone else to take over from here? 🙂

**Comment by alex0711** [comment]
@jankeromnes i won't be able to work on this any time soon - would be great if someone else could continue this part of the documentation :)

File: src/docs/Views.mdx [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+            commandId: OpenDialog.id
+        });
+```
+
+## Using Theia styles
+Running this will work but actually it won't look pleasing. Our text is black and therefore hard to read with the dark theme. Theia is making use of CSS variables to define common colors, fonts and styles. The dark and light theme respectively declare these variables. So by using these common variables, we ensure that our UI looks good in both the dark and light theme (and possible further themes which declare these variables). There are two ways of adding CSS to our DOM elements: either by directly manipulating the DOM or by using a custom stylesheet. We'll use the first approach in this example and the latter one is being used in the view example.
+
+Let's add the base font and color to our root div element:
+```typescript
+    messageNode.setAttribute('style', 'font-family: var(--theia-ui-font-family); color: var(--theia-ui-font-color1);');
```
Comment (akosyakov) [inline src/docs/Views.mdx]: there is no such color styles anymore

**Review by marcdumais-work — CHANGES_REQUESTED** [review]
This repository was recently re-licensed (https://github.com/theia-ide/theia-website/pull/128/commits/441eac118227ed4920b66fe1619122f8a7659d45). I think we would need to obtain permission from this PR's contributor(s) to use/merge it, under the new license. 

cc: @JonasHelming

**Comment by alxflam** [comment]
@marcdumais-work I guess this PR is obsolete with the recent PR #214 which covers pretty much everything i had in my draft here (also my draft is pretty outdated). Agree (then i'd close the PR) or do you want to keep it open?

**Comment by JonasHelming** [comment]
> @marcdumais-work I guess this PR is obsolete with the recent PR #214 which covers pretty much everything i had in my draft here (also my draft is pretty outdated). Agree (then i'd close the PR) or do you want to keep it open?

Thanks for coming back. I believe we can close this, though, as you suggested.

**Comment by alxflam** [comment]
@JonasHelming  :thumbsup: closing.

**Pull request closed without merging**